+nosnoc/+integrator/SmoothedPss.m

```diff
@@ -0,0 +1,174 @@
+classdef SmoothedPss < handle
+    properties
+        model
+        opts
+        integrator_opts
+
+        dcs
+        ode_func
+    end
+
+    properties(Access=private)
+        x_curr
+
+        t_grid
+        t_grid_full
+        x_res
+        x_res_full
+    end
+
+    methods
+        function obj = SmoothedPss(model, opts, integrator_opts)
+            import casadi.*
+            obj.model = model;
+            obj.opts = opts;
+            obj.integrator_opts = integrator_opts;
+
+            opts.preprocess();
+            integrator_opts.preprocess();
+            model.verify_and_backfill(opts);
+
+            if class(model) == "nosnoc.model.Cls" && opts.time_freezing
+                model = nosnoc.time_freezing.reformulation(model, opts);
+                model.verify_and_backfill(opts);
+                obj.model = model;
+            end
+
+            % Run pipeline
+            switch class(model)
+              case "nosnoc.model.Pss"
+                obj.dcs = nosnoc.dcs.Heaviside(model);
+                obj.dcs.generate_variables(opts);
+                obj.dcs.generate_equations(opts);
+
+                sigma = SX.sym('sigma');
+                fun_opts.allow_free = true;
+                f_x_alpha = Function('f_x_alpha', {obj.dcs.alpha}, {obj.dcs.f_x}, fun_opts);
+                f_x_smoothed = f_x_alpha(0.5*(1+tanh([obj.model.c{:}]/sigma)));
+
+                rhs_fun = Function('rhs_fun', {model.x, model.u, sigma}, {f_x_smoothed});
+                obj.ode_func = @(t, x , u ,sigma) full(rhs_fun(x, u, sigma));
+              otherwise
+                error("nosnoc: Incorrect model type for smoothed Pss integrator.")
+            end
+        end
+        
+        function [t_grid,x_res,t_grid_full,x_res_full] = simulate(obj, plugin, extra_args)
+            arguments
+                obj nosnoc.integrator.SmoothedPss
+                plugin nosnoc.solver.MpccMethod = nosnoc.solver.MpccMethod.SCHOLTES_INEQ
```

Comment:
why is this needed here? The smoothed integrator does not solve any MPECs.

Reply:
This is here to maintain interoperability in terms of positional optional arguments with `nosnoc.integrator.FESD.simulate`.

One way around this is to stop passing the `MpccMethod` as a "plugin" and have it be part of `nosnoc.solver.Options` and remove this as an argument. Though actually this in particular can also be removed easily in the integrator namespace PR as we no longer expose this as a public interface.